improver_tests/utilities/test_cube_checker.py

```diff
@@ -182,32 +187,37 @@ def test_coord_promotion_missing_scalar(self):
         """Test case in which a scalar coordinate has been lost from new_cube,
         meaning the cube undergoing checking ends up with different dimension
         coordinates to the progenitor cube. This raises an error."""
-        cube = set_up_cube()
-        new_cube = iris.util.squeeze(cube)
+        new_cube = iris.util.squeeze(self.cube)
         new_cube.remove_coord("realization")
         msg = "The number of dimension coordinates within the new cube"
         with self.assertRaisesRegex(CoordinateNotFoundError, msg):
-            check_cube_coordinates(cube, new_cube)
+            check_cube_coordinates(self.cube, new_cube)
 
 
 class Test_find_dimension_coordinate_mismatch(IrisTest):
 
     """Test if two cubes have the dimension coordinates."""
 
+    def setUp(self):
+        """Set up a cube."""
+        data = np.ones((2, 16, 16), dtype=np.float32)
+        data[:, 7, 7] = 0.0
+        self.cube = set_up_variable_cube(
+            data, "precipitation_amount", "kg m^-2", "equalarea",
+        )
+
     def test_no_mismatch(self):
         """Test if there is no mismatch between the dimension coordinates."""
-        cube = set_up_cube()
-        result = find_dimension_coordinate_mismatch(cube, cube)
+        result = find_dimension_coordinate_mismatch(self.cube, self.cube)
         self.assertIsInstance(result, list)
         self.assertFalse(result)
 
     def test_mismatch_in_first_cube(self):
         """Test when finding a one-way mismatch, so that the second cube has
         a missing coordinate. This returns an empty list."""
-        cube = set_up_cube()
-        first_cube = cube.copy()
-        second_cube = cube.copy()
-        second_cube.remove_coord("time")
+        first_cube = self.cube.copy()
+        second_cube = self.cube.copy()
+        second_cube.remove_coord("realization")
```

Comment:
I don't like this very much because it creates input cubes with an anonymous dimension, which isn't really the situation we'd expect to be seeing when we call this function.  Instead we might expect to be seeing a cube that doesn't have a realization dimension (as opposed to using "remove_coord" here which retains the dimension, but removes its name).  Can I suggest the following design instead?
```
second_cube = next(self.cube.slices_over("realization")).copy()
second_cube.remove_coord("realization")
result = find_dimension_coord_mismatch(
    self.cube, second_cube, two_way_mismatch=False
)
```
And similar for the test(s) below.

Reply:
Done.